### PR TITLE
build(deps): upgrade promql-parser to 0.4 

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Eat the dog food.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by waynexia — APPROVED** [review]
LGTM

File: src/promql/src/planner.rs [hunk 1]
```diff
@@ -641,7 +641,10 @@ impl PromPlanner {
             }
         }
         let matchers = matchers.into_iter().collect();
-        Ok(Matchers { matchers })
+        Ok(Matchers {
```
Comment (yuanbohan) [inline src/promql/src/planner.rs:644]: Maybe `Matchers::new(matchers)`? The `or_matchers` field is only used in [VictoriaMetrics](https://victoriametrics.com/) context, and there is no need to be aware of this field so far.

**Review by yuanbohan — APPROVED** [review]
LGTM

File: src/promql/src/planner.rs (hunk 1 above)
Comment (tisonkun) [reply src/promql/src/planner.rs:644]: Thank you. I made the trivial chanegs to let it compiled. Unaware of `Matchers::new(matchers)` before. Let me update it.

**Review by yuanbohan — APPROVED** [review]

**Pull request merged**